src/hammer-vlsi/par/openroad/__init__.py

```diff
@@ -80,152 +197,1018 @@ def export_config_outputs(self) -> Dict[str, Any]:
+                    bottom=margins.bottom,
+                    right=margins.right,
+                    top=margins.top
+                )
+            else:
+                orientation = constraint.orientation if constraint.orientation is not None else "r0"
+                orientation = orientation.upper()
+                if constraint.create_physical:
+                    pass
+                if constraint.type == PlacementConstraintType.Dummy:
+                    pass
+                elif constraint.type == PlacementConstraintType.Placement:
+                    pass
+                # for OpenROAD
+                elif constraint.type in [PlacementConstraintType.HardMacro, PlacementConstraintType.Hierarchical]:
+                    output.append("place_cell -inst_name {inst} -origin {{ {x} {y} }} -orient {orientation} -status FIRM".format(
+                        inst=new_path,
+                        x=constraint.x,
+                        y=constraint.y,
+                        orientation=orientation,
+                    ))
+                    # TODO: add place_cell option [-status (PLACED|FIRM)]
+                    spacing = self.get_setting("par.blockage_spacing")
+                    if constraint.top_layer is not None:
+                        current_top_layer = constraint.top_layer #  type: Optional[str]
+                    elif global_top_layer is not None:
+                        current_top_layer = global_top_layer
+                    else:
+                        current_top_layer = None
+                    # TODO: find equivalent for place/route halo in OpenROAD
+
+                elif constraint.type == PlacementConstraintType.Obstruction:
+                    pass
+                else:
+                    assert False, "Should not reach here"
+        return [chip_size_constraint] + output
+
+    def specify_std_cell_power_straps(self, blockage_spacing: Decimal, bbox: Optional[List[Decimal]], nets: List[str]) -> List[str]:
+        """
+        Generate a list of TCL commands that build the low-level standard cell power strap rails.
+        This will use the -master option to create power straps based on technology.core.tap_cell_rail_reference.
+        The layer is set by technology.core.std_cell_rail_layer, which should be the highest metal layer in the std cell rails.
+
+        :param bbox: The optional (2N)-point bounding box of the area to generate straps. By default the entire core area is used.
+        :param nets: A list of power net names (e.g. ["VDD", "VSS"]). Currently only two are supported.
+        :return: A list of TCL commands that will generate power straps on rails.
+        """
+        layer_name = self.get_setting("technology.core.std_cell_rail_layer")
+        layer = self.get_stackup().get_metal(layer_name)
+        tcl=[]
+        tcl.append(f"add_pdn_stripe -grid {{grid}} -layer {{{layer.name}}} -width {0.48} -pitch {5.44} -offset 0 -followpins\n")
```

Comment:
Can this width/pitch be gleaned from the stackup?

Reply:
I couldn't figure out how...this 0.48 value doesn't seem to appear anywhere (searched in the LEFs as well). I'll mark this down as another question to ask the openroad developers.